### DISCUSSION TITLE
Allow embedding target in obfuscated URI.

### DISCUSSION THREAD

**Discussion by roke-julian-lockhart** [dubinc/dub#1172] [Ideas]
#### **Request**

I'd like to be able to post a Dub URI, (like `https://dub.sh/bZ0Ke7R`) with its target embedded:

```URI
https://dub.sh/bZ0Ke7R?target=https%3A%2F%2Fbugs.kde.org%2Fshow_bug.cgi%3Fid%3D465948
```

≤ [`LinkStack/releases/tag/v4.6.1`](https://github.com/LinkStackOrg/LinkStack/releases/tag/v4.6.1#:~:text=Releases%20v4.6.1-,v4.6.1,-Compare) provided this functionality [^1] (and is where I gained the idea from): 

[^1]: [`LinkStack/issues/760#issuecomment-2408478520`](https://github.com/LinkStackOrg/LinkStack/issues/760#issuecomment-2408478520)

```URI
https://linksta.cc/going/9358?https://community.frame.work/u/rokejulianlockhart/summary
```

If this were provided as an option when copying the URI from the dashboard into the user's clipboard, it would be perfect (and wouldn't affect anyone who doesn't want it, since it wouldn't be the default).

#### **Rationale**

Summarily, this would help to prevent [link rot](https://en.wikipedia.org/w/index.php?title=Link_rot&oldid=1239627925#firstHeading:~:text=External%20links-,Link%20rot,-27%20languages).

I am aware that `app.dub.co` is primarily a link *shortener*, but it's also as much a link obfuscator, fundamentally. I don't utilize it for either purpose, however - merely as a means to track engagement and allow me a centralized dashboard whereby I can update hyperlinks that have become 404s.

However, in the case that my account were ever removed (by myself or the platform shutting down) I would be seriously disappointed to know that all the Dub URIs I'd posted would become entirely dead hyperlinks.

#### **Alternatives**

You might think that merely appending the target to a fragment (`#`) would serve this purpose, but that has defined functionality - it's not equivalent to a mere comment.

**Comment by linear** [comment]
<p><a href="https://linear.app/dubinc/issue/ENG-502/embed-target-in-obfuscated-uri">ENG-502 Embed target in obfuscated URI.</a></p>

**Comment by roke-julian-lockhart** [comment]
> https://github.com/dubinc/dub/issues/1171#event-13831036516

@steven-tey, why a discussion?

**Comment by roke-julian-lockhart** [comment]
https://github.com/mastodon/mastodon/issues/22424#issuecomment-1764721202 is relevant.
